contracts/extensions/GovernorSettableFixedQuorumUpgradeable.sol

```diff
@@ -0,0 +1,74 @@
+import {Checkpoints} from "@openzeppelin/contracts/utils/structs/Checkpoints.sol";
+
+/// @title GovernorSettableFixedQuorumUpdateable
+/// @author [ScopeLift](https://scopelift.co)
+/// @notice An abstract extension to the Governor which implements a fixed quorum which can be updated by governance.
+/// @custom:security-contact TODO: Add security contact
+abstract contract GovernorSettableFixedQuorumUpgradeable is Initializable, GovernorUpgradeable {
+    using Checkpoints for Checkpoints.Trace224;
+
+    /// @custom:storage-location erc7201:openzeppelin.storage.GovernorSettableFixedQuorum
+    struct GovernorSettableFixedQuorumStorage {
+        Checkpoints.Trace224 _quorumCheckpoints;
+    }
+
+    // keccak256(abi.encode(uint256(keccak256("storage.GovernorSettableFixedQuorum")) - 1)) &
+    // ~bytes32(uint256(0xff))
+    bytes32 private constant GovernorSettableFixedQuorumStorageLocation =
+        0xfc13b0622f15c0c6558fbb10a1f5d36853f903c871b669bae7001d348b50ca00;
+
+    function _getGovernorSettableFixedQuorumStorage()
+        private
+        pure
+        returns (GovernorSettableFixedQuorumStorage storage $)
+    {
+        assembly {
+            $.slot := GovernorSettableFixedQuorumStorageLocation
+        }
+    }
+
+    /// @notice Emitted when the quorum value has changed.
+    event QuorumUpdated(uint256 oldQuorum, uint256 newQuorum);
+
+    function __GovernorSettableFixedQuorum_init(uint224 _initialQuorum) internal onlyInitializing {
+        __GovernorSettableFixedQuorum_init_unchained(_initialQuorum);
+    }
+
+    function __GovernorSettableFixedQuorum_init_unchained(uint224 _initialQuorum) internal onlyInitializing {
+        _setQuorum(_initialQuorum);
+    }
+
+    /// @notice Initializer function to set the initial quorum.
+    /// @param _initialQuorum The number of total votes needed to pass a proposal.
+    function initialize(uint224 _initialQuorum) public initializer {
+        _setQuorum(_initialQuorum);
+    }
+
+    /// @notice A function to set quorum for the current block timestamp. Proposals created after this timestamp will be
+    /// subject to the new quorum.
+    /// @param _amount The new quorum threshold.
+    function setQuorum(uint224 _amount) external onlyGovernance {
+        _setQuorum(_amount);
+    }
+
+    /// @notice A function to get the quorum threshold for a given timestamp.
+    /// @param _voteStart The timestamp of when voting starts for a given proposal.
+    function quorum(uint256 _voteStart) public view override returns (uint256) {
+        GovernorSettableFixedQuorumStorage storage $ = _getGovernorSettableFixedQuorumStorage();
+        return $._quorumCheckpoints.upperLookup(SafeCast.toUint32(_voteStart));
+    }
+
+    /// @notice A function to set quorum for the current block timestamp.
+    /// @param _amount The quorum amount to checkpoint.
+    function _setQuorum(uint224 _amount) internal {
```

Comment:
Also curious why the quorum is now `uint244` instead of `uint256`

Reply:
The Checkpoints library used for storing quorums historically stores the values as uint224 types in an array of structs.

Reply:
I think Checkpoints supports different types. what's the type we actually want here?

Reply:
Looks like Checkpoints supports uint224, uint208, and uint160.  We'd want to stay with 224 (I would think) as it offers the most precision?

Reply:
> Looks like Checkpoints supports uint224, uint208, and uint160. We'd want to stay with 224 (I would think) as it offers the most precision?

I see that all OZ contracts use Trace208 instead of Trace224. 

AI says:

<img width="429" alt="image" src="https://github.com/user-attachments/assets/4532374a-f08e-4808-8a7a-dc7217ffc555">

Reply:
Good point.. and also switched to public quorum of uint256 and internally uint408 types.. I like this because it isolates the choice into one place.. thanks.. good stuff!  Done: d086d70856a6922ee06b33eab608503baf0955a3